frame/staking/src/lib.rs

```diff
@@ -2535,6 +2567,7 @@ impl<T: Trait> Module<T> {
 		}
 
 		// assume the given score is valid. Is it better than what we have on-chain, if we have any?
+		// TODO: errors up to here need refund
```

Comment:
Yeah this can be done for cheap, I will add it.

Reply:
So I have done this, but I will leave it open here because I am now not sure if it is a good idea. Basically these are really cheap checks if we fail here, we revert without doing anything significant. 

I do want to refund here because a whole lot of weight is going to waste. 
I do preferably want to user who caused this to pay for the full fee because this is a critical function. Do I have a way of doing this? Or should we have some manual slashing here?

Reply:
> I do want to refund here because a whole lot of weight is going to waste.
I do preferably want to user who caused this to pay for the full fee because this is a critical function. 

I think user can pay for what he used. I'm not sure slashing is necessary. Anyway both is fine, refund is fine because if user want to pay to fill the block he can do so using any costy extrinsic.
not refund is also fine because the cost of this extrinsic will probably make user careful anyway.

So I'm ok with refund.